### PR TITLE
feat(keploy): adding logging functionality for failed test-cases

### PR DESCRIPTION
**Pull request by Pradhyuman-sharma**
Adding logging functionality for failed test-cases which makes UI optional

### CODE REVIEW

**Comment by Pradhyuman-sharma** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Pradhyuman-sharma** [comment]
_I have read the CLA Document and I hereby sign the CLA_

**Comment by re-Tick** [comment]
LGTM

**Comment by re-Tick** [comment]
Hi @Pradhyuman-sharma, You have to set the username in your local git. You can run this command in your terminal:
`git config --global user.name "FIRST_NAME LAST_NAME"`

**Comment by Pradhyuman-sharma** [comment]
Done

**Comment by slayerjain** [comment]
@Pradhyuman-sharma can you please rebase and fix the merge conflicts? also waiting for your final push

**Comment by Pradhyuman-sharma** [comment]
My work tree is clean ..but rebase showing conflicts of previous commit histories ..Can you please fix it manually ?

**Comment by slayerjain** [comment]
@Pradhyuman-sharma can you please sign your commits for DCO to pass? For instructions, you can just refer to the DCO details page.

File: pkg/match.go [hunk 1]
```diff
@@ -33,6 +33,22 @@ func convertJson(s string, log *zap.Logger) (interface{}, error) {
 
 }
 
+//JsonBody return the expected and actual body in JSON
+func JsonBody(exp, act string, noise []string, log *zap.Logger) (interface{}, interface{}) {
```
Comment (slayerjain) [inline pkg/match.go:37]: we should use more descriptive method name

File: pkg/platform/mgo/rdb.go [hunk 2]
```diff
@@ -90,6 +90,17 @@ func (r *RunDB) PutTest(ctx context.Context, t run.Test) error {
 	return nil
 }
 
+func (r *RunDB) GetTestRun(ctx context.Context, id string) (*run.TestRun, error) {
```
Comment (slayerjain) [inline pkg/platform/mgo/rdb.go:93]: The method name convention should be consistent with other methods in the file.

**Comment by re-Tick** [comment]
Made the changes recommended😃. Please review and merge

**Review by slayerjain — APPROVED** [review]
LGTM

**Pull request merged**